# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1466868997 14400
#      Sat Jun 25 11:36:37 2016 -0400
# Node ID b5e75ce299d8ea32a862b1cc3efc5ee51776c156
# Parent  f6e119d0bdfdb6763739dfa7374d0b5b7cf46ca6
use pattern rule for Bison

* libinterp/module.mk: Cancel suffix rule generated by automake.
(%.cc %.h : %.yy): New pattern rule.
* libinterp/corefcn/module.mk (libinterp/corefcn/oct-tex-parser.h):
Delete dependency on libinterp/corefcn/oct-tex-parser.yy.
* libinterp/parse-tree/module.mk (libinterp/parse-tree/oct-parse.h):
Delete dependency on libinterp/parse-tree/oct-parse.cc.

diff --git a/libinterp/corefcn/module.mk b/libinterp/corefcn/module.mk
--- a/libinterp/corefcn/module.mk
+++ b/libinterp/corefcn/module.mk
@@ -315,17 +315,16 @@ libinterp/corefcn/oct-tex-lexer.ll: libi
 
 libinterp/corefcn/oct-tex-symbols.cc: libinterp/corefcn/oct-tex-symbols.in | libinterp/corefcn/$(octave-dirstamp)
 	$(AM_V_GEN)rm -f $@-t && \
 	$(AWK) 'BEGIN { print "// DO NOT EDIT. AUTOMATICALLY GENERATED FROM oct-tex-symbols.in."; print "static uint32_t symbol_codes[][2] = {"; count = 0; } END { print "};"; printf("static int num_symbol_codes = %d;\n", count); } !/^#/ && (NF == 3) { printf("  { %s, %s },\n", $$2, $$3); count++; }' $< > $@-t && \
 	mv $@-t $@
 
 libinterp/corefcn/txt-eng.cc: libinterp/corefcn/oct-tex-symbols.cc
 libinterp/corefcn/oct-tex-lexer.cc: LEX_OUTPUT_ROOT := lex.octave_tex_
-libinterp/corefcn/oct-tex-parser.h: libinterp/corefcn/oct-tex-parser.yy
 
 libinterp/corefcn/oct-tex-parser.yy: libinterp/corefcn/oct-tex-parser.in.yy
 	$(AM_V_GEN)$(call subst-bison-api-decls,octave_tex_)
 
 noinst_LTLIBRARIES += \
   libinterp/corefcn/libcorefcn.la
 
 libinterp_corefcn_libcorefcn_la_SOURCES = $(COREFCN_SRC)
diff --git a/libinterp/module.mk b/libinterp/module.mk
--- a/libinterp/module.mk
+++ b/libinterp/module.mk
@@ -207,21 +207,21 @@ DLL_CXXDEFS = @OCTINTERP_DLL_DEFS@
 LIBINTERP_TST_FILES_SRC := $(shell $(SHELL) $(srcdir)/build-aux/find-files-with-tests.sh "$(srcdir)" $(ULT_DIST_SRC) $(DLDFCN_SRC))
 
 LIBINTERP_TST_FILES := $(addsuffix -tst,$(LIBINTERP_TST_FILES_SRC))
 
 libinterptestsdir := $(octtestsdir)
 
 nobase_libinterptests_DATA = $(LIBINTERP_TST_FILES)
 
-## Override Automake's rule that forces a .hh extension on us even
-## though we don't want it.  It would be super awesome if automake
-## would allow users to choose the header file extension.
+## Cancel the suffix rule and use a pattern rule instead.
 .yy.cc:
-	$(AM_V_YACC)$(am__skipyacc) $(SHELL) $(YLWRAP) $< y.tab.c $@ y.tab.h $*.h y.output $*.output -- $(YACCCOMPILE)
+
+%.cc %.h : %.yy
+	$(AM_V_YACC)$(am__skipyacc) $(SHELL) $(YLWRAP) $< y.tab.c $*.cc y.tab.h $*.h y.output $*.output -- $(YACCCOMPILE)
 
 ## Special rules:
 ## Mostly for sources which must be built before rest of compilation.
 
 libinterp/build-env.cc: libinterp/build-env.in.cc build-aux/subst-config-vals.sh | libinterp/$(octave-dirstamp)
 	$(AM_V_GEN)$(call simple-filter-rule,build-aux/subst-config-vals.sh)
 
 libinterp/build-env-features.cc: config.h libinterp/build-env-features.sh | libinterp/$(octave-dirstamp)
diff --git a/libinterp/parse-tree/module.mk b/libinterp/parse-tree/module.mk
--- a/libinterp/parse-tree/module.mk
+++ b/libinterp/parse-tree/module.mk
@@ -81,18 +81,16 @@ PARSE_TREE_SRC = \
 ## oct-gperf.h file.
 libinterp/parse-tree/oct-gperf.h: libinterp/parse-tree/octave.gperf
 	$(AM_V_GEN)rm -f $@-t $@t1 $@ && \
 	$(GPERF) -t -C -D -G -L C++ -Z octave_kw_hash $< > $@-t1 && \
 	$(SED) -e 's,lookup\[,gperf_lookup[,' -e 's,register ,,g' < $@-t1 > $@-t && \
 	mv $@-t $@ && \
 	rm -f $@-t1
 
-libinterp/parse-tree/oct-parse.h: libinterp/parse-tree/oct-parse.cc
-
 libinterp/parse-tree/oct-parse.yy: libinterp/parse-tree/oct-parse.in.yy
 	$(AM_V_GEN)$(call subst-bison-api-decls,octave_)
 
 noinst_LTLIBRARIES += \
   libinterp/parse-tree/libparse-tree.la
 
 libinterp_parse_tree_libparse_tree_la_SOURCES = $(PARSE_TREE_SRC)
 
